MozillaFoundation/foundation.mozilla.org
# Add "wide" option to article TableBlock

# Description

This PR will create a new `WideTableBlock` on article pages with a checkbox so that users can select a table width of either wider than page content (checked by default), or the same width as page content (unchecked by default).

The editor experience remains the same, except the addition of now having a checkbox to change the table width.

Related PRs/issues: TP1-144 / #10332 

# To Test
This is a little tricky because I was only able to find privately published examples of basic tables on articles.

1) Check out `main`
2) Copy staging data via `inv copy-stage-db`
3) Login to the wagtail dashboard via `/cms/`
4) Navigate to the parent public page: http://localhost:8000/cms/pages/16712/edit/
5) Change visibility to `Public` and publish the page
6) Navigate to the view the table on the article page: http://localhost:8000/en/campaigns/personalised-pricing/personalised-pricing/
8) Navigate to the article edit page in order to note the editor screen of the table with no checkbox: http://localhost:8000/cms/pages/16721/edit/
9) Check out this branch `TP1-144-10332-add-wide-table-block` (you may also need a new stage db)
10) Run migrations via `inv migrate`
11) Navigate to http://localhost:8000/en/campaigns/personalised-pricing/personalised-pricing/ in order to confirm the migrated table looks the same
12) Navigate to http://localhost:8000/cms/pages/16721/edit/ in order to view the editor screen of the table and confirm the `Wide` option now exists and is checked by default.
13) Uncheck "Wide" to reduce the width to the content of the page and publish the page.
14) Stop docker and re-run `docker-compose up` in order to get new tailwind classes
15) Navigate to http://localhost:8000/en/campaigns/personalised-pricing/personalised-pricing/ in order to view the now narrow table.

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1193)
